### ISSUE TITLE
Add support for threads count

### ISSUE THREAD

**Issue by simonepri** [soyuka/pidusage#57]
Example:
```bash
❯ ps M -p 2100        
USER        PID   TT   %CPU STAT PRI     STIME     UTIME COMMAND
simonepri  2100   ??    1.9 R    47T   0:01.72   0:18.31 /Applications/Utilities/Terminal.app/Contents/MacOS/Terminal
           2100         0.0 S    20T   0:00.00   0:00.00 
           2100         0.0 S    31T   0:00.00   0:00.00 
           2100         0.0 S    31T   0:00.54   0:00.33 
           2100         0.0 S    47T   0:00.92   0:00.56 
           2100         0.1 S    61T   0:00.00   0:00.03 
           2100         0.1 S    47T   0:00.00   0:00.02 
           2100         0.0 S    61T   0:00.00   0:00.01 
           2100         0.1 S    47T   0:00.00   0:00.01 
```

To count them:
```bash
❯ ps M -p 2100 | wc -l
      10
```

@soyuka any idea on how can we support this?

**Comment by simonepri** [comment]
ps on AIX has the thcount column.
https://superuser.com/a/49411/470946

**Comment by simonepri** [comment]
proc files have the Thread value
http://ask.xmodulo.com/number-of-threads-process-linux.html

**Comment by soyuka** [comment]
Nice research! I agree this could be a nice addition!

**Comment by simonepri** [comment]
Can't find a nice way to do it with `ps`, though.

**Comment by soyuka** [comment]
thcount seems available on linux

**Comment by simonepri** [comment]
Cool so the problem is only macOS :(

**Comment by soyuka** [comment]
Yes :| https://stackoverflow.com/questions/28047653/osx-how-can-i-see-the-tid-of-all-threads-from-my-process

**Comment by soyuka** [comment]
I guess we could introduce this and set a default number for os x while we find a way to do this properly on os x.

**Comment by simonepri** [comment]
On osx we need a custom ps command and parsing function.
I'll try to work on it as soon as my exam session ends.

**Comment by soyuka** [comment]
that's cool! 

Could you check on os x if you have:
`/proc/{pid}/tasks` (number of files, maybe faster then reading status) or `/proc/{pid}/status` (search the file for `Threads`) ?

**Comment by simonepri** [comment]
On macOS there's no /proc folder sadly.
I'll take a look on how cpuUsage() works under the hood in node, for sure there's some particular syscall to call to obtain the same info.

**Comment by soyuka** [comment]
Not sure that they actually use the number of threads :|. We won't be able to make this work without bindings.

**Comment by hiyelbaz** [comment]
Would you consider adding this feature for Linux and Windows only? Related function can return -1 on macOS which would indicate that the platform is not supported.
